pyof/foundation/basic_types.py

```diff
@@ -412,7 +412,7 @@ class BinaryData(GenericType):
     return the size of the instance using Python's :func:`len`.
     """
 
-    def __init__(self, value=b''):  # noqa
+    def __init__(self, value=None):  # noqa
```

Comment:
Accepting None will be useful to change body's default to None. Please, add a test for this behavior.